### PR TITLE
Chore: Move cache item to lib

### PR DESCRIPTION
**Pull request by iLLiCiTiT**
## Changelog Description
We have implemented multiple different approaches how to cache data, usually the same. Goal of this PR is to have one in `ayon_core.lib` that can be used across the board. Using the items in `Anatomy` as first step.

## Additional info
The classes were taken from `ayon_core.tools.common_models.cache` where were used for model data. There are 2 classes, one for storing the data, second for nesting them in specific level. The nesting was usually done using lambda functions.

## Testing notes:
1. Anatomy caching still works.
2. Validate implementation of the classes.

### CODE REVIEW

File: client/ayon_core/lib/cache.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+
+def _default_factory_func():
+    return None
+
+
+class CacheItem:
+    """Simple cache item with lifetime and default factory for default value.
+
+
```
Comment (BigRoy) [inline client/ayon_core/lib/cache.py:17]:
```suggestion
```

**Comment by BigRoy** [comment]
> ## Testing notes:
> 
>     1. Anatomy caching still works.

Do you have some details on how one could best test that?

**Comment by iLLiCiTiT** [comment]
> Do you have some details on how one could best test that?

That's a very good question. I did it by printing `frame_padding` in templates each second which I changed while it was printing.

File: client/ayon_core/lib/cache.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+    Default factory should return default value that is used on init
+        and on reset.
+
+    Args:
+        default_factory (Optional[callable]): Function that returns default
+            value used on init and on reset.
+        lifetime (Optional[int]): Lifetime of the cache data in seconds.
```
[the remark this answers is not in the document]
Comment (iLLiCiTiT) [reply client/ayon_core/lib/cache.py:23]: Added to docstring. It should handle the `None` because of `NestedCacheItem` anyway.

**Pull request merged**